src/form/textarea/Textarea.tsx

```diff
@@ -0,0 +1,153 @@
+import "./_textarea.scss";
+
+import React, {useState} from "react";
+import classNames from "classnames";
+import TextareaAutosize, {TextareaAutosizeProps} from "react-textarea-autosize";
+
+import {KEYBOARD_EVENT_KEY} from "../../core/utils/keyboardEventConstants";
+
+type TextareaProps = Omit<
+  React.TextareaHTMLAttributes<HTMLTextAreaElement>,
+  "name" | "onChange" | "className"
+> & {
+  name: string;
+  onChange: React.ReactEventHandler<HTMLTextAreaElement>;
+  value?: string;
+  testid?: string;
+  isDisabled?: boolean;
+  customClassNames?: {
+    container?: string;
+    textarea?: string;
+  };
+  style?: React.CSSProperties;
+  autoSizeProps?: TextareaAutosizeProps;
+  onJustEnterPressed?: () => void;
+  onShiftEnter?: () => void;
+  isRequired?: boolean;
+  hasError?: boolean;
+};
+
+function Textarea(props: TextareaProps) {
+  const {
```

Comment:
Let's try to clean this here and only destructure the variables that are necessary. If all the variables are required to destructure, you can ignore this however as far as I can see we can remove some of them and spread `otherProps` when passing props to the elements below